### PR TITLE
feat: [PLG-2330]: save card and validate via subscription flow

### PR DESCRIPTION
**Pull request by dhruvUpadhyay07**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/33-auth-settings/components/Subscription/PaymentMethod/PaymentMethodStep.tsx [hunk 1]
```diff
@@ -70,30 +108,74 @@ export default function PaymentMethodStep({
...
+              }}
+              checked={useExistingCard}
+              className={css.useExistingCard}
+            >
+              <PaymentMethodCard
+                paymentMethodInfo={subscriptionProps.paymentMethodInfo}
+                setView={setView}
+                module={module}
+              />
+              {/* <PaymentMethod viaCreditCardFlow={true} creditCardData={fetchedCreditCardData} /> */}
```
Comment (devinder-harness) [inline src/modules/33-auth-settings/components/Subscription/PaymentMethod/PaymentMethodStep.tsx:158]: commented code

File: src/modules/33-auth-settings/components/Subscription/PaymentMethod/Footer.tsx [hunk 2]
```diff
@@ -25,102 +25,142 @@ interface FooterProps {
...
-        })
+          //save the card details call the save api
+
+          //save the card details call the save api
+          await saveCard({
+            accountIdentifier: accountId,
+            creditCardIdentifier: res?.paymentMethod?.id as string
+          })
+        }
+        showSuccess('card is validated and saved')
```
Comment (devinder-harness) [inline src/modules/33-auth-settings/components/Subscription/PaymentMethod/Footer.tsx:146]: move to strings?

File: src/modules/33-auth-settings/components/Subscription/PaymentMethod/PaymentMethodStep.tsx [hunk 3]
```diff
@@ -70,30 +105,73 @@ export default function PaymentMethodStep({
...
+            </Radio>
+          </Layout.Horizontal>
+          <Radio
+            large
+            onChange={() => {
+              setAddCard(true)
+              setUseExistingCard(false)
+            }}
+            label={getString('common.addNewCreditCard')}
+            data-testid={`add new card`}
```
Comment (devinder-harness) [inline src/modules/33-auth-settings/components/Subscription/PaymentMethod/PaymentMethodStep.tsx:164]: test id with `space`?

**Review by devinder-harness — DISMISSED** [review]

**Review by kapil-harness — DISMISSED** [review]

File: src/modules/33-auth-settings/components/Subscription/BillingInfo/__test__/PaymentMethodStep.test.tsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+const clientSecret = 'dummy secret'
+const stripePromise = loadStripe('dummy promise')
+
+jest.mock('services/cd-ng')
+
+const useGetDefaultCardMock = useGetDefaultCard as jest.MockedFunction<any>
+useGetDefaultCardMock.mockImplementation(() => {
+  return {
+    data: {
+      correlationId: '40d39b08-857d-4bd2-9418-af1aafc42d20',
```
Comment (mayankVermaHarness) [inline src/modules/33-auth-settings/components/Subscription/BillingInfo/__test__/PaymentMethodStep.test.tsx:27]:
```suggestion
      correlationId: 'dummy',
```

File: src/modules/33-auth-settings/components/Subscription/BillingInfo/__test__/PaymentMethodStep.test.tsx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        </Elements>
+      </TestWrapper>
+    )
+
+    expect(container).toMatchSnapshot('payment step  method  getting rendered')
+    // await userEvent.click(getByText('back'))
+    // await waitFor(() => {
+    //   expect(setLoadPaymentMethodStep).toBeCalled()
+    //   expect(setViewMock).toBeCalledWith(SubscribeViews.PAYMENT_METHOD)
+    // })
```
Comment (mayankVermaHarness) [inline src/modules/33-auth-settings/components/Subscription/BillingInfo/__test__/PaymentMethodStep.test.tsx:98]: Remove if not needed.

File: src/modules/33-auth-settings/components/Subscription/FinalReview/PaymentMethodCard.tsx [hunk 6]
```diff
@@ -37,7 +37,7 @@ const PaymentMethodCard: React.FC<PaymentMethodCardProps> = ({ paymentMethodInfo
                 category: Category.CREDIT_CARD,
                 module
               })
-              setView(SubscribeViews.PAYMENT_METHOD)
+              if (setView !== undefined) setView(SubscribeViews.PAYMENT_METHOD)
```
Comment (mayankVermaHarness) [inline src/modules/33-auth-settings/components/Subscription/FinalReview/PaymentMethodCard.tsx:40]: Nit: 
```suggestion
                setView?.(SubscribeViews.PAYMENT_METHOD)
```

File: src/modules/33-auth-settings/components/Subscription/PaymentMethod/Footer.tsx [hunk 7]
```diff
@@ -25,102 +25,142 @@ interface FooterProps {
...
-          city,
-          country,
-          creditCardId: res.paymentMethod?.id,
-          line1: billingAddress,
-          state,
-          zipCode
-        })
+          //save the card details call the save api
+
+          //save the card details call the save api
```
Comment (mayankVermaHarness) [inline src/modules/33-auth-settings/components/Subscription/PaymentMethod/Footer.tsx:140]: double comment

File: src/modules/33-auth-settings/pages/Billing/BillingPage.module.scss [hunk 8]
```diff
@@ -175,6 +176,10 @@
   }
 }
 
+.loading {
+  margin-left: 50px;
```
Comment (mayankVermaHarness) [inline src/modules/33-auth-settings/pages/Billing/BillingPage.module.scss:180]: Can we using `--spacing` variables here?

File: src/modules/33-auth-settings/strings/strings.en.yaml [hunk 9]
```diff
@@ -285,7 +285,7 @@ billingInfo:
   paymentMethod: Payment Method
   paymentMethods: Payment Methods
   billingAdmin: '{{$.common.billing}} Admin'
-  saveCardWarning: By providing your card information, you allow Harness Inc. to charge your card for future payments in accordance with our Terms of Use.
+  saveCardWarning: By checking this box,you allow Harness Inc. to charge your card and save it for the future module license charges in accordance with our Terms of Use.
```
Comment (mayankVermaHarness) [inline src/modules/33-auth-settings/strings/strings.en.yaml:288]:
```suggestion
  saveCardWarning: By checking this box, you allow Harness Inc. to charge your card and save it for the future module license charges in accordance with our Terms of Use.
```

File: src/modules/33-auth-settings/strings/strings.en.yaml [hunk 10]
```diff
@@ -362,3 +362,4 @@ yesIamSure: Yes, I am very sure
 jitProvisioning: JIT Provisioning
 jitValidationKey: JIT Validation Key
 jitValidationValue: JIT Validation Value
+cardIsValidatedAndSaved: card is validated and saved
```
Comment (mayankVermaHarness) [inline src/modules/33-auth-settings/strings/strings.en.yaml:365]:
```suggestion
cardIsValidatedAndSaved: Card is validated and saved
```

**Review by devinder-harness — APPROVED** [review]

**Review by kapil-harness — APPROVED** [review]

**Review by mayankVermaHarness — APPROVED** [review]

**Review by kapil-harness — DISMISSED** [review]

**Review by devinder-harness — DISMISSED** [review]

**Review by tan-nhu — DISMISSED** [review]

**Comment by dhruvUpadhyay07** [comment]
retrigger cypress-pipeline

**Review by tan-nhu — APPROVED** [review]

**Review by kapil-harness — APPROVED** [review]

**Pull request merged**